File: JavaTest/StringBuilderTest.java
import MyLibrary.*;
import org.junit.*;

import static org.junit.Assert.*;

public class StringBuilderTest extends TestBase {
    @Test
    public void AsciiTest() {
        String asciiValue = "Some value";
        StringBuilderExample sb = new StringBuilderExample(asciiValue);
        assertEquals(asciiValue, sb.CreateString());
    }

    @Test
    public void UnicodeTest() {
        String unicodeValue = "吝吞吟吠吡";
        StringBuilderExample sb = new StringBuilderExample(unicodeValue);
        assertEquals(unicodeValue, sb.CreateString());
    }
}

File: CSharpTest/StringBuilderTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CSharpTest
{
    using SB = MyLibrary.StringBuilderExample;

    [TestClass]
    public class StringBuilderTest
    {
        [TestMethod]
        public void AsciiTest()
        {
            string asciiValue = "Some value";
            var sb = new SB(asciiValue);
            Assert.AreEqual(asciiValue, sb.CreateString());
        }

        [TestMethod]
        public void UnicodeTest()
        {
            // Known to fail with SWIGWin 3.0.5
            string unicodeValue = "吝吞吟吠吡";
            var sb = new SB(unicodeValue);
            Assert.AreEqual(unicodeValue, sb.CreateString());
        }
    }
}
